consensus/istanbul/core/commit.go

```diff
@@ -96,6 +97,16 @@ func (c *core) handleCommit(msg *message, src istanbul.Validator) error {
 
 	c.acceptCommit(msg, src)
 
+	// Change to Prepared state if we've received enough PREPARE or COMMIT messages and we are in earlier state
+	// before Prepared state.
+	// Both of PREPARE and COMMIT messages are counted since hashlocked nodes in the previous round skip sending
+	// PREPARE messages.
+	if c.current.GetPrepareOrCommitSize() > 2*c.valSet.F() && c.state.Cmp(StatePrepared) < 0 {
+		c.current.LockHash()
+		c.setState(StatePrepared)
```

Comment:
If 2f+1 commits arrive first, shouldn't the committed also be set? Is it set in other logic?

Reply:
@yoomee1313 If 2f+1 commits arrived, the node will commit the proposal in the next lines. It means that the state will be changed to `StateCommitted`. 
https://github.com/klaytn/klaytn/blob/604c761dc94c7d16bb0a2d0867e24435187c292d/consensus/istanbul/core/commit.go#L122-L126
Do I understand your question correctly?

Reply:
Oh there it is, thank you!